### DISCUSSION TITLE
Anti Cheat Protection

### DISCUSSION THREAD

**Discussion by o777l** [winstxnhdw/lc-hax#431] [General]
Last night I ended up in a lobby with a host using some kind of anti cheat mod, which was advertised in the chat.

As soon as someone joined using minimap, modified shovel damage, infinite ammo, triggering landmines etc. the host was notified of this in chat.

I showed up as well, but it was not apparent to me what it detected. It only said in red text "◻◻◻◻◻◻◻ o777I 3/4"

This is the anti cheat used https://github.com/chuxiaaaa/AntiCheat

Now another question, if anyone wants to test this mod and lc-hax only  in some lobbies to see if it works or not, I'm up for it. Just message me

**Comment by winstxnhdw** [comment]
The square texts are Chinese. Without knowing the exact, characters, I won’t be able to know what did they detect. Seems like it detects anti-kick though.

**Reply by Totoqoe** [reply]
I joined a lobby once with antikick that had this mod and didn't get kicked, though an important detail is that I took someone's profile so it might have bypassed the detection since it only detects the name it seems.

**Comment by winstxnhdw** [comment]
The anti-cheat is pretty good.

```cs
        private static bool KillPlayerServerRpc(NetworkBehaviour target, FastBufferReader reader, __RpcParams rpcParams)
        {
            if (Check(rpcParams, out var p))
            {
                var e = (EnemyAI)target;
                ByteUnpacker.ReadValueBitPacked(reader, out int playerId);
                reader.Seek(0);
                AntiCheatPlugin.ManualLog.LogInfo($"{p.playerUsername} call {e.GetType()}.KillPlayerServerRpc|playerId:{playerId}");
                if (playerId <= StartOfRound.Instance.allPlayerScripts.Length && !StartOfRound.Instance.allPlayerScripts[playerId] != p)
                {
                    AntiCheatPlugin.ManualLog.LogInfo($"{p.playerUsername} call {e.GetType()}.KillPlayerServerRpc|playerUsername:{StartOfRound.Instance.allPlayerScripts[playerId].playerUsername}");
                    return false;
                }
            }
            else if (p == null)
            {
                return false;
            }
            return true;
        }
```

It can detect who creates the RPC call. I wonder if I can use this to spoof RPC calls.

**Reply by o777l** [reply]
Let me know if you need help testing.

**Reply by Dexteriority** [reply]
So the summary of what currently detected from lc-hax are Anti-Kick & KillCommand ?

**Reply by winstxnhdw** [reply]
No idea. I wonder how they get around false positives though. For example, if a eyeless dog is chasing me and my teammate runs into the eyeless dog, I think that would count as a false positive and the mod will kick me lmao

**Reply by Dexteriority** [reply]
lol, thats undestandable since its still an experimental mod

**Comment by o777l** [comment]
> So the summary of what currently detected from lc-hax are Anti-Kick & KillCommand ?

I forgot to add that I wasn't using any commands or anti kick that particular session, wanted to see if I was detected after injecting.

I don't really have any other mods that are considered cheats in my mod list

**Reply by winstxnhdw** [reply]
Interesting.

**Comment by winstxnhdw** [comment]
Can someone help test if they get kicked if you switch off the lights manually vs using `/light`?

**Reply by o777l** [reply]
I could host a game with the anti cheat if you want, my code is 1560224837

**Reply by winstxnhdw** [reply]
I am not on my computer right now, so I can’t test it personally.

**Reply by Dexteriority** [reply]
Lets do it

**Reply by Dexteriority** [reply]
@o777l I added you on steam, lets test it out

**Reply by o777l** [reply]
@Dexteriority Just saw this, I'm on now

**Comment by winstxnhdw** [comment]
I’d also like someone to test pulling the lever for the ship to leave vs using `/end`.
